plugins/modules/vcenter_root_password_expiration.py

```diff
@@ -0,0 +1,156 @@
+
+
+class VcRootPasswordExpiration(VmwareRestClient):
+    def __init__(self, module: AnsibleModule) -> None:
+        self.module = module
+        self._state = True if self.module.params['state'] == 'present' else False
+
+    def configure_root_account_password_policy(self):
+        default_config = self.api_client.appliance.LocalAccounts.UpdateConfig()
+
+        current_vcenter_info = self.api_client.appliance.LocalAccounts.get('root').to_dict()
+
+        if self._state and self.module.params['min_days_between_password_change'] > self.module.params['max_days_between_password_change']:
+            self.module.fail_json("min_days_between_password_change cannot be higher than max_days_between_password_change")
+
+        if self._state:
+            _password_expiration_config = {
+                "email": self.module.params['email'],
+                "min_days_between_password_change": self.module.params['min_days_between_password_change'],
+                "max_days_between_password_change": self.module.params['max_days_between_password_change'],
+                "warn_days_before_password_expiration": self.module.params['warn_days_before_password_expiration'],
+
+            }
+        else:
+            _password_expiration_config = {
+                "max_days_between_password_change": -1
+            }
+
+        _changes_dict = dict()
+        for k, v in _password_expiration_config.items():
+            try:
+                if current_vcenter_info[k] != v:
+                    _changes_dict[k] = v
+                if k == 'fullname':
+                    setattr(default_config, 'full_name', v)
+                    continue
+            except KeyError:
+                '''
+                Handles the case of newly installed vCenter when email field isn't present in the current config,
+                because it was never set befores
+                '''
+                _changes_dict[k] = v
+            setattr(default_config, k, v)
+        _change_result_key = 'values_would_be_changed'
+        if _changes_dict:
+            if not self.module.check_mode:
+                _change_result_key = 'values_changed'
+                self.api_client.appliance.LocalAccounts.update('root', default_config)
+            self.module.exit_json(changed=True, result={_change_result_key: _changes_dict})
+        self.module.exit_json(changed=False, result="No configuration changes needed")
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        dict(
+            state=dict(default='present',
+                       choices=['present', 'absent'],
+                       type='str')
+        ),
+        email=dict(required=False, type='str'),
+        max_days_between_password_change=dict(type='int', no_log=True),
+        min_days_between_password_change=dict(type='int', no_log=True),
+        warn_days_before_password_expiration=dict(type='int', no_log=True),
```

Comment:
Are you sure you _don't_ want to log this? I don't think this is really sensitive information (like a password). But not logging them might make it hard for people to troubleshoot, especially if they're using variables that are defined in several places.

[Variable precedence](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#understanding-variable-precedence) is quite complex. It's easy to make a mistake there (I've done it often enough myself) and very hard to troubleshoot if you don't see the value in the logs.

Reply:
I also don't want to not log the variables, but the sanity check is failing because the no-log is missing?
```
plugins/modules/vcenter_root_password_expiration.py:0:0: no-log-needed: Argument 'max_days_between_password_change' in argument_spec could be a secret, though doesn't have no_log set
plugins/modules/vcenter_root_password_expiration.py:0:0: no-log-needed: Argument 'min_days_between_password_change' in argument_spec could be a secret, though doesn't have no_log set
plugins/modules/vcenter_root_password_expiration.py:0:0: no-log-needed: Argument 'warn_days_before_password_expiration' in argument_spec could be a secret, though doesn't have no_log set
```

Reply:
@valentinJonev I'm not 100% sure, but I think the sanity tests assume these are sensitive parameters because they contain "password". At lease, it's a possibility. This way, parameters like `new_password`, `root_password` or similar would be matched.

However, the [documentation](https://docs.ansible.com/ansible/latest/dev_guide/testing/sanity/validate-modules.html#codes) says you can stop the test from complaining by explicitly setting `no_log=False`:

```
Option name suggests that the option contains a secret value, while no_log is not specified for this option in the argument spec. If this is a false positive, explicitly set no_log=False
```